internal/service/service_others.go

```diff
@@ -13,16 +13,22 @@
 // limitations under the License.
 
 //go:build !windows
-// +build !windows
 
-package main
+package service
 
 import (
-	"context"
+	"os"
+	"os/signal"
+	"syscall"
 
-	"go.opentelemetry.io/collector/service"
+	"go.uber.org/zap"
 )
 
-func run(ctx context.Context, params service.CollectorSettings) error {
-	return runInteractive(ctx, params)
+// RunService runs the given service, calling its start and stop functions.
+func RunService(logger *zap.Logger, rSvc RunnableService) error {
+	stopSignal := make(chan os.Signal, 1)
+	signal.Notify(stopSignal, syscall.SIGINT, syscall.SIGTERM)
```

Comment:
Curious why we didn't want to use the signal.NotifyContext but rather use a channel?

Reply:
Good point, changed this to a context.